lib/config/flat-config-schema.js

```diff
@@ -98,16 +87,21 @@ function deepMerge(first, second = {}, mergeMap = new Map()) {
     }
 
     /*
-     * First create a result object where properties from the second value
+     * First create a result object where properties from the second object
      * overwrite properties from the first. This sets up a baseline to use
      * later rather than needing to inspect and change every property
      * individually.
      */
-    const result = {
+    let result = {
         ...first,
         ...second
     };
 
+    delete result.__proto__; // eslint-disable-line no-proto -- don't merge own property "__proto__"
```

Comment:
If we don't want the `__proto__`, then I'd suggest using `Object.create(null)` to create `result`.

Reply:
I think the intent here was just to exclude `__proto__` when it's an _own_ property, because the inherited property `Object.prototype.__proto__` should not be masked or accidentally overwritten (`delete` only removes own properties). By contrast, `Object.create(null)` creates an object with `null` prototype, that lacks all methods of `Object.prototype` but could still have an own property called `__proto__`, so it's not really the same thing.

Reply:
Alternatively, we could treat `__proto__` as a regular property with the usual merge behavior as other properties. We'd just need to take care to use `Object.defineProperty` rather than a regular assignment (`=`) to set `__proto__` in order to not mess up with the prototype ([here](https://github.com/eslint/eslint/blob/8d84b0b94fa8a8b2d868d08ff20cf8df14a24dbb/lib/config/flat-config-schema.js#L119) and [here](https://github.com/eslint/eslint/blob/8d84b0b94fa8a8b2d868d08ff20cf8df14a24dbb/lib/config/flat-config-schema.js#L121)).

Reply:
Ah gotcha, so this line is deleting an own property that might have been added through the spread operations. That makes sense to me. :+1: